### PR TITLE
[POC] setup turbo repo

### PR DESCRIPTION
**Pull request by vitalikda**
# Description

POC: monorepo to Turbo repo 

- `cli` folder is not affected by the changes
- `packages` folder is where grafbase npm packages and eslint/ts configs are located
- `cli` and `packages` workflows are set up to run only when respective folders have changes
- ~~changeset/cli is setup to manage releases~~

# Type of change

- [ ] 💔 Breaking
- [x] 🚀 Feature
- [ ] 🐛 Fix
- [x] 🛠️ Tooling
- [ ] 🔨 Refactoring
- [ ] 🧪 Test
- [x] 📦 Dependency
- [ ] 📖 Requires documentation update

### CODE REVIEW

File: pnpm-workspace.yaml [hunk 1]
```diff
@@ -0,0 +1,3 @@
+packages:
+  - "apps/*"
```
Comment (fbjork) [inline pnpm-workspace.yaml:2]: apps?

Comment (vitalikda) [reply pnpm-workspace.yaml:2]: PR is WIP 
apps/ will be removed (used it to test nextjs-grafbase)

Comment (vitalikda) [reply pnpm-workspace.yaml:2]: @notrab I refactored your [POC](https://github.com/notrab/next-grafbase/), feel free to review here: [/packages/grafbase-nextjs-plugin](https://github.com/grafbase/grafbase/tree/gb-2713-setup-nextjs-grafbase-plugin/packages/grafbase-nextjs-plugin)

File: .eslintrc.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
+module.exports = {
+  root: true,
+  // This tells ESLint to load the config from the package `eslint-config-grafbase`
```
Comment (fbjork) [inline .eslintrc.js:3]: is this comment needed when this is expected behavior?

File: .eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
...
+  extends: ['grafbase'],
+  settings: {
+    next: {
+      rootDir: ['apps/*/']
```
Comment (fbjork) [inline .eslintrc.js:7]: should be packages?

File: packages/grafbase-nextjs-plugin/package.json [hunk 4]
```diff
@@ -0,0 +1,31 @@
+{
```
Comment (fbjork) [inline packages/grafbase-nextjs-plugin/package.json:1]: should the folder be nextjs-plugin or grafbase-nextjs-plugin?

Comment (vitalikda) [reply packages/grafbase-nextjs-plugin/package.json:1]: File naming can be anything, but it's recommended to use prefix to identify public packages from internal ones

File: .eslintrc.js (hunk 3 above)
Comment (vitalikda) [reply .eslintrc.js:7]: This can be removed, this is a hack for next eslint plugin issue

File: .eslintrc.js (hunk 2 above)
Comment (vitalikda) [reply .eslintrc.js]: Removed

File: .changeset/late-chefs-swim.md [hunk 5]
```diff
@@ -0,0 +1,5 @@
+---
```
Comment (fbjork) [inline .changeset/late-chefs-swim.md:1]: what's with the filename?

File: packages/grafbase-nextjs-plugin/package.json (hunk 4 above)
Comment (fbjork) [reply packages/grafbase-nextjs-plugin/package.json:1]: What does internal ones mean?

**Comment by yoav-lavi** [comment]
@vitalikda have you checked this with required actions in the branch protection settings?

File: .changeset/late-chefs-swim.md (hunk 5 above)
Comment (yoav-lavi) [reply .changeset/late-chefs-swim.md:1]: Looks like one of those random generators that create memorable names by using dictionary words like for initial project names etc.

File: packages/grafbase-nextjs-plugin/package.json (hunk 4 above)
Comment (vitalikda) [reply packages/grafbase-nextjs-plugin/package.json:1]: packages with 'grafbase-' prefix are published to npm, the rest like eslint ts configs, are using only inside workspace

File: .changeset/late-chefs-swim.md (hunk 5 above)
Comment (vitalikda) [reply .changeset/late-chefs-swim.md:1]: This is temporary file, which is used to track changesets, will be removed when package is published and content will be added to CHANGELOG.md 
Example: https://github.com/grafbase/grafbase/pull/143/files

**Comment by vitalikda** [comment]
> @vitalikda have you checked this with required actions in the branch protection settings?

Yes, I have, I used cli workflow as a reference.
Note that I am not able to change setting for required jobs, so if we merge this PR, someone will need to make packages jobs required as well.

I used paths filter initially, but for required jobs there is a workaround that I had to make for them to not stay in stalled/pending state, here is the doc: [troubleshooting-required-status-checks](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/troubleshooting-required-status-checks#handling-skipped-but-required-checks)

However, it complicated the workflows, so latest changes use [paths-filter action](https://github.com/dorny/paths-filter/#example), which conditionally skips jobs – "Skipped" status treated as "passed" by github ci.

**Comment by yoav-lavi** [comment]
@vitalikda looks like this still skips the CLI when it's changed and adds a few changesets to CLI changes:
https://github.com/grafbase/grafbase/pull/147

**Comment by yoav-lavi** [comment]
Not sure why but I got a changset here when committing to the CLI: https://github.com/grafbase/grafbase/pull/147
 testing whether breaking the CLI still prevents committing now

**Comment by yoav-lavi** [comment]
@vitalikda from what I can tell changing the CLI to break it causes the packages actions to run but not the CLI ones:
https://github.com/grafbase/grafbase/pull/148

File: package.json [hunk 6]
```diff
@@ -0,0 +1,23 @@
+{
```
Comment (yoav-lavi) [inline package.json:1]: Not sure we need this in the root, would be a bit weird for it to work on anything other than CLI. Maybe we should group the other folders under a project?

File: pnpm-workspace.yaml [hunk 7]
```diff
@@ -0,0 +1,2 @@
+packages:
```
Comment (yoav-lavi) [inline pnpm-workspace.yaml:1]: Same here and for `turbo.json`, unless we plan to use turborepo for the CLI in some capacity

**Review by notrab — APPROVED** [review]
:shipit:

**Pull request merged**